engine/database/user_utxos.go

```diff
@@ -0,0 +1,15 @@
+package database
+
+import "time"
+
+// UserUtxos is a table holding user's Unspent Transaction Outputs (UTXOs).
+type UserUtxos struct {
```

Comment:
yes this is another dimension of output, and when recording transaction and the output became the input, it should be removed (deleted) from this table (for all users - in case if we will handle multi ownership of the output)